File: tic-tac-toe-web/src/main/java/by/xxx/pupil/model/GameRequest.java
package by.xxx.pupil.model;

import java.util.Arrays;
import java.util.Objects;

public class GameRequest {

    private final char[][] board;
    private final Cell playerMove;
    private final Integer difficultyLevel;

    public GameRequest(char[][] board, Cell playerMove, Integer difficultyLevel) {
        this.board = board;
        this.playerMove = playerMove;
        this.difficultyLevel = difficultyLevel;
    }

    public char[][] getBoard() {
        return board;
    }

    public Cell getPlayerMove() {
        return playerMove;
    }

    public Integer getDifficultyLevel() {
        return difficultyLevel;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        GameRequest that = (GameRequest) o;
        return difficultyLevel.equals(that.difficultyLevel) &&
                Arrays.equals(board, that.board) &&
                Objects.equals(playerMove, that.playerMove);
    }

    @Override
    public int hashCode() {
        int result = Objects.hash(playerMove, difficultyLevel);
        result = 31 * result + Arrays.hashCode(board);
        return result;
    }
}

File: tic-tac-toe-web/src/main/kotlin/by/pupil/model/GameRequest.kt
package by.pupil.model

import java.util.*

data class GameRequest(val board: Array<Array<Char>>, val playerMove: WebCell, val difficultyLevel: Int) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as GameRequest
        return difficultyLevel == that.difficultyLevel &&
            board.contentEquals(that.board) &&
            playerMove == that.playerMove
    }

    override fun hashCode(): Int {
        var result = Objects.hash(playerMove, difficultyLevel)
        result = 31 * result + Arrays.hashCode(board)
        return result
    }
}
